src/main/webapp/js/captions.js

```diff
@@ -376,6 +402,132 @@ function setClickableEntities() {
     });
 }
 
+/**
+ * Fetches and displays the bookmarks for the current user and video
+ * @param email - The email of the current user
+ * @param videoId - The videoId of the current video
+ */
+function fetchBookmarks(email, videoId) {
+    var fetchUrlBuilder = '/bookmark?email=' + email + '&videoId=' + videoId;
+
+    fetch(fetchUrlBuilder).then(response => response.json()).then(json => {
+        $('#bookmark-display-div').html('');
+        if (Array.isArray(json)) {
+            displayBookmarks(json);
+        } else if (typeof json === 'object' && json !== null) {
+            console.log(json.ERROR);
+            alert(json.ERROR);
+        }
+    });
+
+}
+
+/**
+ * Renders bookmarks in HTML from a list of Bookmark objects
+ * @param list - The list of Bookmark objects
+ */
+function displayBookmarks(list) {
+    // Resets the global bookmarks variable to only store current bookmarks
+    bookmarks = {};
+
+    // Builds the HTML text to display on page
+    var output = '';
+    for (bookmark of list) {
+        bookmarks[bookmark.id] = {'timestamp': bookmark.timestamp, 'content': bookmark.content};
```

Comment:
Nit: This function is a little too long. Let's create a function called ```renderBookmark(bookmark)``` to extract the rendering logic out.

Reply:
will address in `tab-searching` branch, which has more front end designs